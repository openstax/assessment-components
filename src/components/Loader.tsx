import { BulletList } from "react-content-loader";
import { colors } from "../../src/theme";
import styled from "styled-components";

const StyledLoader = styled(BulletList)`
  max-width: 1000px;
  padding: 2rem;
  border: 1px solid ${colors.palette.light};
  border-radius: 0.25rem;
  background-color: white;
`;

export const Loader = () => <StyledLoader />;
