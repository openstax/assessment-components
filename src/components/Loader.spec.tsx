import renderer from 'react-test-renderer';
import { Loader } from './Loader';

describe('Loader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Loader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
