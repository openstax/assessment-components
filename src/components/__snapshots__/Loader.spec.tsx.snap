// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<svg
  aria-labelledby="q5irv9-aria"
  className="sc-bczRLJ dtqrhK"
  id={null}
  role="img"
  style={Object {}}
  viewBox="0 0 245 125"
>
  <title
    id="q5irv9-aria"
  >
    Loading...
  </title>
  <rect
    clipPath="url(#q5irv9-diff)"
    height="100%"
    role="presentation"
    style={
      Object {
        "fill": "url(#q5irv9-animated-diff)",
      }
    }
    width="100%"
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="q5irv9-diff"
    >
      <circle
        cx="10"
        cy="20"
        r="8"
      />
      <rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="15"
      />
      <circle
        cx="10"
        cy="50"
        r="8"
      />
      <rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="45"
      />
      <circle
        cx="10"
        cy="80"
        r="8"
      />
      <rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="75"
      />
      <circle
        cx="10"
        cy="110"
        r="8"
      />
      <rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="105"
      />
    </clipPath>
    <linearGradient
      id="q5irv9-animated-diff"
    >
      <stop
        offset="0%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-2; -2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#eee"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-1; -1; 2"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="0; 0; 3"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
